### PR TITLE
Vue: move to Vitest

### PR DESCRIPTION
**Pull request by qmonmert**
Fix: #3831

TODO (coverage):
image

it seems that separated vue file and component file, breaks coverage...

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Is it ready @qmonmert ? Or still todo about coverage?

File: src/main/resources/generator/client/vue/vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    logHeapUsage: true,
+    minThreads: 1,
+    maxThreads: 2,
+    environment: 'jsdom',
+    cache: false,
+    include: ['src/test/javascript/spec/**/*.(spec|test).(ts|tsx)'],
+    exclude: ['node_modules', 'src/test/javascript/integration/**/*.spec.ts'],
+    coverage: {
+        clean: true,
+        provider: 'istanbul',
```
Comment (pascalgrimaud) [inline src/main/resources/generator/client/vue/vitest.config.ts:27]: Fix ident to 2 spaces plz

**Comment by qmonmert** [comment]
> Is it ready @qmonmert ? Or still todo about coverage?

locally there is always a line uncovered

**Comment by pascalgrimaud** [comment]
I confirm 2 things:
- the uncovered line, but I don't understand why
- the full build using mvn clean verify doesn't fail, and it should fail

**Comment by pascalgrimaud** [comment]
Similar to what was done in https://github.com/jhipster/jhipster-lite/pull/3868, I don't know if it's a good exclude, but adding this part seems to solve the issue:

image

**Comment by qmonmert** [comment]
CI error because one line is not covered

**Pull request merged**